go/runtime/enclaverpc/api/api.go

```diff
@@ -3,8 +3,16 @@ package api
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/oasislabs/oasis-core/go/common"
+	"github.com/oasislabs/oasis-core/go/common/cbor"
+)
+
+const (
+	// Make sure this always matches the appropriate method in
+	// `keymanager-runtime/src/methods.rs`.
+	GetPublicKeyRequestMethod = "get_public_key"
```

Comment:
any recommendations on how to handle this? are you fine with moving this (and the other methods) back to keymanager package, but still using it in `enclaverpc` service authentication definition?

Reply:
No, the check is specific to the key manager enclave RPC endpoint, so it should not be part of generic EnclaveRPC. In this case the access control function should be defined here:
https://github.com/oasislabs/oasis-core/blob/ptrus/feature/grpc-sentry-support/go/worker/keymanager/service.go

What is the reason that the access control function needs to be defined in the generic gRPC service definition?

Reply:
So another question -- how will this work in this example:
* If the worker is a key manager, it exports an EnclaveRPC service connecting to the key manager enclave on the external gRPC server. It uses access control policy A.
* If the worker is a foobar worker, it exports an EnclaveRPC service connecting to the foobar enclave on the external gRPC server. It uses access control policy B.

Each instance should have its own access control policy.

Reply:
> What is the reason that the access control function needs to be defined in the generic gRPC service definition?

so that the proxy (`sentry/grpc`) could apply the correct access control. Without it, we cannot really know from the request itself if policies should be applied or not.

> So another question -- how will this work in this example:

maybe we shouldn't have a "generic" enclaberpc service?

Reply:
i guess we could extend the `WatchPolicies` thing that the sentry uses, to also support sharing custom (non "grpc/policy")  policies. But i think this "other" type of policies should first be a bit generalized, so it can be reused between different services.

Reply:
> maybe we shouldn't have a "generic" enclaverpc service?

So the idea is that you can call arbitrary methods via EnclaveRPC regardless of endpoints, e.g., that the API is the same so that you can use the same client.

I would be ok with having multiple server-side instances (even with different gRPC service names and different access control policies), but the methods should be exactly the same and should map to the `Transport` interface.

So the service name could be `EnclaveRPC.KeyManager` and the km-specific bits would be defined in the km package. It would be nice if we could avoid duplicating all the definitions as other than the access control policy they would be exactly the same.

Maybe in this case it would make sense to have a factory of EnclaveRPC gRPC services? So basically you could just wrap `serviceName`, `method*`, `serviceDesc` into a structure and have `handler*` be methods. Then you could have a constructor that creates a new service and you could pass access control policies?

Reply:
yeah, this approach makes sense, thanks :+1:

Reply:
done something like this ^

Reply:
done something like this ^